core/src/main/java/nl/nn/adapterframework/receivers/PullingListenerContainer.java

```diff
@@ -115,7 +115,12 @@ public boolean isThreadCountControllable() {
 
 	@Override
 	public int getCurrentThreadCount() {
-		return (int)threadsRunning.getValue();
+		log.trace("Get current thread-count, synchronized (lock) on threadsRunning[{}]", threadsRunning);
+		try {
+			return (int)threadsRunning.getValue();
+		} finally {
+			log.trace("Get current thread-count, lock on threadsRunning[{}] released", threadsRunning);
+		}
```

Comment:
De `IThreadCountControllable` methodes lijken mij een beetje overdreven. Die worden volgens mij alleen aangeroepen vanuit de console.

Reply:
@nielsm5 De value wordt alleen vanuit de console opgevraagd, maar opgehoogd en verlaagd tijdens de verwerking van een bericht. Als je trace logging wil weghalen, dan alleen rondom het opvragen en dan zou ik het liever doen in 7.9 eerst en dan backporten naar 7.8/7.7.
Dus ik stel voor dat te doen als we een deadlock vinden of anders volgende keer dat we in master iets moeten doen met deze code?